### PR TITLE
Remove long <-> ulong casts from Nvservices code

### PR DESCRIPTION
**Pull request by macabeus**
Closes https://github.com/Ryujinx/Ryujinx/issues/1820

It's my first PR on this project and my first time coding in C#.
Let me know if there is something that I could improve, please.

### CODE REVIEW

File: Ryujinx.HLE/HOS/Services/SurfaceFlinger/SurfaceFlinger.cs [hunk 1]
```diff
@@ -308,11 +308,11 @@ private void PostFrameBuffer(Layer layer, BufferItem item)
                 nvMapHandle = item.GraphicBuffer.Object.Buffer.NvMapId;
             }
 
-            int bufferOffset = item.GraphicBuffer.Object.Buffer.Surfaces[0].Offset;
+            ulong bufferOffset = (ulong)item.GraphicBuffer.Object.Buffer.Surfaces[0].Offset;
 
             NvMapHandle map = NvMapDeviceFile.GetMapFromHandle(layer.Owner, nvMapHandle);
 
-            ulong frameBufferAddress = (ulong)(map.Address + bufferOffset);
+            ulong frameBufferAddress = map.Address + bufferOffset;
```
Comment (macabeus) [inline Ryujinx.HLE/HOS/Services/SurfaceFlinger/SurfaceFlinger.cs:315]: I know that this file is outside of `Nv` service, but I needed to update these two lines too.

File: Ryujinx.HLE/HOS/Services/Nv/NvDrvServices/NvHostAsGpu/Types/AddressSpaceContext.cs [hunk 2]
```diff
@@ -12,54 +12,54 @@ private class Range
...
+                Start = position;
+                End   = size + Start;
             }
         }
 
         private class MappedMemory : Range
         {
-            public long PhysicalAddress { get; private set; }
+            public ulong PhysicalAddress { get; private set; }
             public bool VaAllocated     { get; private set; }
```
Comment (gdkchan) [inline Ryujinx.HLE/HOS/Services/Nv/NvDrvServices/NvHostAsGpu/Types/AddressSpaceContext.cs:25]: nit: Alignment.

**Review by gdkchan — COMMENTED** [review]
Looks good other than that single comment I left.

**Review by Thog — APPROVED** [review]
lgtm, same comment as gdkchan.
Thanks again 👍

File: Ryujinx.HLE/HOS/Services/Nv/NvDrvServices/NvHostAsGpu/Types/AddressSpaceContext.cs (hunk 2 above)
Comment (macabeus) [reply Ryujinx.HLE/HOS/Services/Nv/NvDrvServices/NvHostAsGpu/Types/AddressSpaceContext.cs]: done

**Comment by macabeus** [comment]
@Thog Thanks. Done

**Review by gdkchan — APPROVED** [review]
lgtm, thanks.

**Pull request merged**